src/tests/ftest/io/eight_servers_mpiio.py

```diff
@@ -24,77 +24,35 @@
 from __future__ import print_function
 
 import os
-import sys
-import json
-from avocado import Test
-
-sys.path.append('./util')
-sys.path.append('../util')
-sys.path.append('../../../utils/py')
-sys.path.append('./../../utils/py')
-import server_utils
 import write_host_file
 import ior_utils
+
+from apricot import TestWithServers
 from mpio_utils import MpioUtils, MpioFailed
-from daos_api import DaosContext, DaosPool, DaosApiError
+from daos_api import DaosPool, DaosApiError
 
-class EightServers(Test):
+class EightServers(TestWithServers):
     """
     Test class Description: Runs IOR with 8 servers.
 
     """
 
-    def __init__(self, *args, **kwargs):
-
-        super(EightServers, self).__init__(*args, **kwargs)
+    def setUp(self):
+        super(EightServers, self).setUp()
 
-        self.basepath = None
-        self.server_group = None
-        self.context = None
-        self.pool = None
-        self.num_procs = None
-        self.hostlist_servers = None
-        self.hostfile_servers = None
-        self.hostlist_clients = None
-        self.hostfile_clients = None
+        # set required variables
         self.mpio = None
-
-    def setUp(self):
-        # get paths from the build_vars generated by build
-        with open('../../../.build_vars.json') as build_file:
-            build_paths = json.load(build_file)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-        print("<<{}>>".format(self.basepath))
-        self.server_group = self.params.get("name", '/server_config/',
-                                            'daos_server')
-
-        # setup the DAOS python API
-        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
-
-        self.hostlist_servers = self.params.get("test_servers",
-                                                '/run/hosts/test_machines/*')
-        self.hostfile_servers = (
-            write_host_file.write_host_file(self.hostlist_servers,
-                                            self.workdir))
-        print("Host file servers is: {}".format(self.hostfile_servers))
-
-        self.hostlist_clients = self.params.get("test_clients",
-                                                '/run/hosts/test_machines/*')
         self.num_procs = self.params.get("np", '/run/ior/client_processes/*')
         self.hostfile_clients = (
```

Comment:
This shouldn't be needed should it, since:
https://github.com/daos-stack/daos/blob/55993845b4b19a2e3d7315a2de9bb77f1cf2b9ed/src/tests/ftest/util/apricot/apricot/test.py#L202-L204

Reply:
It is as we don't want to specify any slots for this test where as by default it's set to '1'. hence mentioned as 'None'

Reply:
Perhaps rather than re-writing the host file, apricot needs extending to define the slots the host file should have when it's created.  The number of slots could be defined in the super class's __init__() if the super class wants to override the default value, for example.

The code referenced above then can look for the _number-of-slots_ attribute in the class and use that if specified or the default of 1 if not.

Reply:
Done